modin/core/storage_formats/base/query_compiler.py

```diff
@@ -132,7 +131,23 @@ def default_to_pandas(self, pandas_op, *args, **kwargs):
         BaseQueryCompiler
             The result of the `pandas_op`, converted back to ``BaseQueryCompiler``.
         """
-        pass
+        op_name = getattr(pandas_op, "__name__", str(pandas_op))
+        ErrorMessage.default_to_pandas(op_name)
+        args = (a.to_pandas() if isinstance(a, type(self)) else a for a in args)
+        kwargs = {
+            k: v.to_pandas() if isinstance(v, type(self)) else v
+            for k, v in kwargs.items()
+        }
```

Comment:
let's use more comprehensive approach 
```suggestion
        args = try_cast_to_pandas(args)
        kwargs = try_cast_to_pandas(kwargs)
```

https://github.com/modin-project/modin/blob/1a57a542ad0413fae2591d4b4928b652effd01db/modin/utils.py#L480